src/Components/Gallery.js

```diff
@@ -0,0 +1,66 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import '../Styles/Gallery.css';
+import { BASE_URL } from '../apis/api';
+
+const Gallery = ({ images, setImages }) => {
+  const [pages, setPages] = useState(1);
+  const [totalPages, setTotalPages] = useState(null);
```

Comment:
Why is your initial value `null`? Is `totalPages` not supposed to hold only integers?

Reply:
I would change it now